Fixing commented out unit tests for EditPostPublishSettingsJetpackSocialUiStateMapperTest

develric opened the pull request:
This PR aims to fix the commented-out unit tests in #18823.

To test:
- Review the unit test code
- Make sure the CI is green

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Restoring existent unit tests

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:

- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

develric committed:
Fixing commented out unit tests for EditPostPublishSettingsJetpackSocialUiStateMapperTest

develric committed:
Refactoring to fixing linter too long function issue.

RenanLukas commented on WordPress/src/test/java/org/wordpress/android/ui/posts/EditPostPublishSettingsJetpackSocialUiStateMapperTest.kt:
```diff
@@ -1,171 +1,183 @@
+        )
+        val shareMessage = "Message"
+        val subscribeButtonLabel = "SHARE MORE"
+        whenever(stringProvider.getString(R.string.post_settings_jetpack_social_subscribe_share_more))
```
💡 Would it make sense to extract this part into a separate method, so it can be used by other tests mocking it instead of duplicating the mocking logic?

e.g.
```
private fun mockGetStringSubscribeShareMore() {
         whenever(stringProvider.getString(R.string.post_settings_jetpack_social_subscribe_share_more))
}
```

RenanLukas commented on WordPress/src/test/java/org/wordpress/android/ui/posts/EditPostPublishSettingsJetpackSocialUiStateMapperTest.kt:
```diff
@@ -1,171 +1,183 @@
+        val subscribeButtonLabel = "SHARE MORE"
+        whenever(stringProvider.getString(R.string.post_settings_jetpack_social_subscribe_share_more))
+            .thenReturn(subscribeButtonLabel)
+        whenever(localProvider.getAppLocale())
```
💡 Would it make sense to extract this part into a separate method, so it can be used by other tests mocking it instead of duplicating the mocking logic?

e.g.
```
private fun mockLocale() {
       whenever(localProvider.getAppLocale())
}
```

RenanLukas reviewed (approved):
Thanks for taking care of this, @develric! I've left a couple of optional suggestions, but overall LGTM :shipit:

develric committed:
Extracting mocking method for string resource.

develric commented:
Thanks @RenanLukas 🙇 . I extracted  `mockStringResource` and moved the `whenever(localProvider.getAppLocale()).thenReturn(Locale.US)` init in the setUp method since it seems to be always the same when needed (commit [here](eb82db8326cc630cfb3ade342475511d79b3df29)) . Let me know if this works, in that case I'll merge this one 🙇

RenanLukas reviewed (approved):
LGTM :shipit:
